cli/proc_state.rs

```diff
@@ -412,6 +438,14 @@ impl ProcState {
         .await?;
       self.prepare_node_std_graph().await?;
     }
+    // If there's a lock file, update it with all discovered npm packages
+    if let Some(lockfile_mutex) = &self.lockfile {
+      let mut lockfile = lockfile_mutex.lock();
+      if let Err(err) = self.npm_resolver.lock(&mut lockfile) {
```

Comment:
Maybe we should store the lockfile in `NpmResolution` then do this on every `npm_resolver.add_package_reqs` call?

Reply:
We could, however I'm not sure if that's preferable. Why do you think it's better to do it in such way?

Reply:
So that we always check the lockfile when adding packages. For example, right now we don't check it in workers, which I think is a bug.

Reply:
Sure, sounds good. I'll rework it